src/vcpkg/tools.cpp

```diff
@@ -79,7 +79,7 @@ namespace vcpkg
 
     Optional<ToolData> parse_tool_data_from_xml(StringView XML, StringView XML_PATH, StringView tool, StringView os)
     {
-        static const std::string XML_VERSION = "2";
+        static const char* XML_VERSION = "2";
```

Comment:
Doesn't this result in literal pointer comparison below, since `match_xml_version[1]` is also a `const char*`? How does this work?

Reply:
match_xml_version[1] is a `sub_match`, which defines https://en.cppreference.com/w/cpp/regex/sub_match/operator_cmp